torch/csrc/autograd/variable.h

```diff
@@ -535,29 +525,19 @@ inline Variable make_variable(
       !data.is_variable(),
       "Must not create a new variable from a variable, use its .tensor_data()");
   if (data.defined()) {
-    auto data_impl_copy = data.getIntrusivePtr()->shallow_copy_and_detach(
-      /*version_counter=*/0,
-      /*allow_tensor_metadata_change=*/allow_tensor_metadata_change);
-    data_impl_copy->set_autograd_meta(c10::guts::make_unique<Variable::AutogradMeta>(
-      data_impl_copy.get(), requires_grad));
-    return Variable(data_impl_copy);
-  }
-  return Variable();
-}
-
-inline Variable make_variable_consuming(
-    at::Tensor data,
-    bool requires_grad = false,
-    bool allow_tensor_metadata_change = true) {
-  TORCH_CHECK(
-      !data.is_variable(),
-      "Must not create a new variable from a variable, use its .tensor_data()");
-  if (data.defined()) {
-    AT_ASSERT(data.getIntrusivePtr().use_count() == 1);
-    auto data_impl = data.getIntrusivePtr();
-    data_impl->set_allow_tensor_metadata_change(allow_tensor_metadata_change);
-    data_impl->set_autograd_meta(c10::guts::make_unique<Variable::AutogradMeta>(data_impl.get(), requires_grad));
-    return Variable(std::move(data_impl));
+    if (data.getIntrusivePtr().use_count() == 1) {
```

Comment:
Isn't this unsafe in a multithreaded environment? Isn't it possible to have two threads invoking `make_variable(tensor)` concurrently on the same `tensor` instance?

Reply:
Wouldn't two threads have to move() the same instance into the call to make_variable? Or is there another sequencing you're thinking of?

Reply:
Or more directly: i believe this is safe because `data` is taken by value

Reply:
Ah yeah, you're right - taking by value makes this safe. The interleaving I was thinking of two threads invoking make_variable(t); one threads passes the use_count check tries to move, and the other thread fails the use_count check and tries to copy from the moved-from value, but that can't happen since neither could pass the use_count check.